README.md

```diff
@@ -53,6 +53,15 @@ fn main() {
     for &value in m.iter() {
         println!("{}", value);
     }
+
+    // Matrix multiplication
+    let m4: Dimensional<i32, LinearArrayStorage<i32, 2>, 2> = matrix![
+        [7, 8],
+        [9, 10],
+        [11, 12]
+    ];
+    let product = m.mat_mul(&m4);
```

Comment:
Question, do you think a trait implementation overloading `*` would be more clear here?

Reply:
It could be, but given that there is already * for elementwise multiplication how to differentiate elementwise vs matrix multiplication for square matrices?

Reply:
Yeah, that's a question on what the preferred API is. As a user, do you find it more intuitive to have the simple operators be element wise and the more complicated operators be function calls? I think it might be, given that the linalg stuff can be gated off then on a crate feature flag from the core array functionality as needed.

If you did want to overload the operator, * could mean element wise between a n=0/scalar and n <= 1 and then could define operations between say 2 vectors as vector multiplication, between two matrices as a dot product. It's something I had wanted to put a little though into for the cleanest API. This is still fluid of course.

Reply:
Cool, as a user I would prefer a different operator for elementwise vs dot product (for instance * elementwise vs @ dot product in Python), but I think there isn't another meaningful operator in Rust to overload. Not sure how to conditionally overload. 

In any case in https://github.com/warlock-labs/dimensionals/pull/4/commits/291de5b521d46768db8c23e9dacb5b761e27c36a if keeping * for simple and elementwise only at least the function is updated to dot to make the syntax more consistent with Python and C++.

Reply:
Great.